README.md
## 说明  
目前这个项目定为新版B/S产品的前端框架项目。这个项目遵循目前流行的互联网前端开发风格进行。  

This project aims to bootstrap the development of your product and provide an ecosystem for building production-ready application or prototypes.

Frameworks like Bootstrap provide a number of components, but usually it’s not enough to build a real-world app. This template comes with lots of popular UI components with a unified color scheme, plus it is based on a modern Angular framework and has a flexible component based structure.  

## 内部网站  
* RAP:api文档和mock [http://vosung.bgenius.cn:8081](http://vosung.bgenius.cn:8081)  
* vosung技术社区 [http://vosung.bgenius.cn:8082/](http://vosung.bgenius.cn:8082/)


## 自动部署（ci）  
本项目采用gitlab-ci实现编码到测试部署的持续集成。   
自动部署的地址是： [http://vosung.bgenius.cn:4200](http://vosung.bgenius.cn:4200)  

RestAPI的内网测试地址：[http://192.168.2.238:8000/metadata](http://192.168.2.238:8000/metadata)   


## 目录  
* 开发框架说明[本页]  
* [API说明](ourdocs/design/api.md)    
* [前端处理逻辑说明](ourdocs/design/frontendlogic.md)  
* [组织管理菜单结构设计](ourdocs/design/Menus/组织管理菜单结构设计.jpg)
* [组织管理原型设计](ourdocs/design/PrototypeProject/Net4FrmPrototype.rar)
* [团队协作Bug榜](ourdocs/mgmtdaily.md)
* [框架开发流程(黄亮)](ourdocs/design/DevelopmentFlow(HL).md)
* [表格组件需求](http://vosung.bgenius.cn/framework-group/ng2-admin/wikis/home)
* [标准组件库](ourdocs/standcomponent.md)

## List of features
* Angular 2+
* Angular CLI
* Bootstrap 4 CSS Framework
* SASS
* Smart Table
* Forms
* Editors
* Charts (amChart, Chartist, Chart.js, Morris)
* Maps (Google, Leaflet, amMap)
* Responsive layout
* High resolution
* and many more!
* 企业组织管理  

## 先决条件  
* [下载安装git](https://git-for-windows.github.io/)  
* [下载安装nodejs(v6.10.2)](https://nodejs.org)  
* [下载安装Python(Python 2.7.13 - 2016-12-17)](https://www.python.org/ftp/python/2.7.13/python-2.7.13.amd64.msi)
* npm install webpack-merge -g  
* npm install webpack -g  
* npm install rimraf -g  

## 运行环境  
框架作者要求：  
Make sure you have Node version >= 6.0 and NPM >= 3  
我们统一支持：  
node v6.10.2  
npm 4.1.2    

## 运行项目  
#### 克隆仓库并且安装依赖  
你可以使用git命令行克隆仓库：  
```
git clone https://github.com/akveo/ng2-admin.git  
```
克隆好仓库之后，进入项目目录，并安装依赖：  
```
cd ng2-admin
npm install
```
or  
```
cd ng2-admin
yarn
```
注意：如果安装依赖失败，可以删除node-modules目录然后重试，如果多次失败则向同事请求协助。  

#### 运行本地副本  
在development mode运行本地副本：  
```
npm start
```  
Go to http://0.0.0.0:4200 or http://localhost:4200 in your browser.  

To run the local copy in production mode and build the sources, execute:  
```
npm run start:prod
```
or in [AOT mode](ourdocs/ngaot.md)    
```
npm run start:prod:aot
```
To create a bundle in production mode, execute:  
```
npm run build:prod
```
or  
```
npm run build:prod:aot
```
This will clear up your dist folder (where release files are located), generate a release build and start the built-in server. Now you can copy the sources from the dist folder and use it with any backend framework or simply put it under a web server.


## Electron Hybird架构  
使用Electron Hybird架构嵌入本Web App的时候，需要对项目下/src/index.html文件进行少许改进。要在页面head节内增加：  
```
<!--这里用来适配Electron : begin  -->
    <script>
        window.nodeRequire = require;
        delete window.require;
        delete window.exports;
        delete window.module;
    </script>
<!--这里用来适配Electron : end  -->
```
相关文章：  
[Electron基础 - 解决无法使用jQuery/RequireJS/Meteor/AngularJS 的问题](https://zhuanlan.zhihu.com/p/21440362)    
Electron Hybird项目地址：http://ascode@vosung.bgenius.cn/framework-group/Net4Frm-Hybird.git   


## 组件/模块/工具说明
* webpack-merge  
* rimraf  
* [Karma](ourdocs/karma.md)  
Karma 是Google 开源的一个基于Node.js 的 JavaScript 测试执行过程管理工具（Test Runner）。该工具可用于测试所有主流Web浏览器，也可集成到 CI （Continuous integration）工具，也可和其他代码编辑器一起使用。  
* mockjs  
[https://github.com/nuysoft/Mock/wiki](https://github.com/nuysoft/Mock/wiki)  
* express路由 
[http://www.expressjs.com.cn/guide/routing.html](http://www.expressjs.com.cn/guide/routing.html)    
* Postman  
api测试工具  


## 帮助
* [Angularjs2 在线教程](http://www.hubwiz.com/course/5599d367a164dd0d75929c76/)  
  
* [bootstrap 4 官方网站](https://v4-alpha.getbootstrap.com/)  

* [奚琛发起的《开发人员帮助手册》(目前唯一的内部开发人员帮助手册，包括前后端开发流程指导，git源码使用指导等等)](http://vosung.bgenius.cn/xic/Devlopment-Helper)  

* [通知公告](http://vosung.bgenius.cn/ascode/notice)  
  
* [前端界面开发框架admui](http://www.admui.com/)  

* [前端界面开发框架H+](http://www.zi-han.net/theme/hplus/index.html)  

* 关于ng serve  
ng serve/ng serve --prod --aot启动项目出现的问题.  
参见 https://github.com/angular/angular-cli/issues/4895 解决。具体原因，其实那该黄色到警告已经说明，angular-cli 已经被抛弃，转到 @angular/cli。所以当我卸载 angular-cli  ，运行npm install -g @angular/cli，成功后就可以用 cli 的命令了。


## 知识储备  

这里将用到的前端知识：  

程序员必须运用的前端知识：  




---
# 如下是作者的源码首页  

[![Build Status](https://travis-ci.org/akveo/ng2-admin.svg?branch=master)](https://travis-ci.org/akveo/ng2-admin)
[![Join the chat at https://gitter.im/ng2-admin/Lobby](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/ng2-admin/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
[![Dependency Status](https://david-dm.org/akveo/ng2-admin/status.svg)](https://david-dm.org/akveo/ng2-admin)

# Admin panel framework based on Angular, Bootstrap 4 and Angular CLI

Admin template made with :heart:  by [Akveo team](http://akveo.com/). Follow us on [Twitter](https://twitter.com/akveo_inc) to get the latest news about this template first!

### Demo

<a target="_blank" href="http://akveo.com/ng2-admin/"><img src="http://i.imgur.com/QK9AzHj.jpg" width="600" alt="Sky Blue"/></a>

<a target="_blank" href="http://akveo.com/ng2-admin/">Live Demo</a>

## Angular 1.x version
Here you can find Angular 1.x based version: [Blur Admin](http://akveo.github.io/blur-admin/)
 
## Documentation
Installation, customization and other useful articles: https://akveo.github.io/ng2-admin/

## Based on
Angular 2+, Angular CLI, Bootstrap 4, and lots of awesome modules and plugins

## BrowserStack
This project runs its tests on multiple desktop and mobile browsers using [BrowserStack](http://www.browserstack.com).

<img src="https://cloud.githubusercontent.com/assets/131406/22254249/534d889e-e254-11e6-8427-a759fb23b7bd.png" height="40" />

## How can I support developers?
- Star our GitHub repo :star:
- Create pull requests, submit bugs, suggest new features or documentation updates :wrench:
- Follow us on [Twitter](https://twitter.com/akveo_inc) :feet:
- Like our page on [Facebook](https://www.facebook.com/akveo/) :thumbsup:

## Can I hire you guys?
Yes!  Visit [our homepage](http://akveo.com/) or simply leave us a message to [contact@akveo.com](mailto:contact@akveo.com). We will be happy to work with you!

## Features
* TypeScript
* Webpack
* Responsive layout
* High resolution
* Bootstrap 4 CSS Framework
* Sass
* Angular
* jQuery
* Charts (Chartist, Chart.js)
* Maps (Google, Leaflet, amMap)
* and many more!

## License
[MIT](LICENSE.txt) license.

### From akveo

Enjoy :metal:
We're always happy to receive your feedback!


CHANGELOG.md
### 变更记录  
* 2017/5/24
变更：  
user数据增加department_unique_id，prd_department_name
涉及： 
数据库表： N4_SysDatUser  
前端： “新增用户”、“修改用户”，“新增角色”，“修改角色”


---

<a name="1.0.0"></a>
# [1.0.0](https://github.com/akveo/ng2-admin/compare/v0.9.0...v1.0.0) (2017-04-27)


### Bug Fixes

* **build:** fix [#777](https://github.com/akveo/ng2-admin/issues/777) ([9bd7dbd](https://github.com/akveo/ng2-admin/commit/9bd7dbd))
* **build:** fix [#955](https://github.com/akveo/ng2-admin/issues/955) ([#961](https://github.com/akveo/ng2-admin/issues/961)) ([dd63535](https://github.com/akveo/ng2-admin/commit/dd63535))
* **build:** hopefully fix the prod & aot build errors ([#753](https://github.com/akveo/ng2-admin/issues/753)) ([7215b0b](https://github.com/akveo/ng2-admin/commit/7215b0b))
* **card:** fix [#962](https://github.com/akveo/ng2-admin/issues/962) ([#966](https://github.com/akveo/ng2-admin/issues/966)) ([72cb319](https://github.com/akveo/ng2-admin/commit/72cb319))
* **components:** add css pseudo-selector for dropdown arrow  ([#848](https://github.com/akveo/ng2-admin/issues/848)) ([1be434b](https://github.com/akveo/ng2-admin/commit/1be434b))
* **components:** add notification and message count from array length ([#893](https://github.com/akveo/ng2-admin/issues/893)) ([a8fe3f1](https://github.com/akveo/ng2-admin/commit/a8fe3f1))
* **dependencies:** fix [#689](https://github.com/akveo/ng2-admin/issues/689), [#688](https://github.com/akveo/ng2-admin/issues/688) ([d34e6e9](https://github.com/akveo/ng2-admin/commit/d34e6e9))
* **dependencies:** fix angular2-template-loader version ([2b25258](https://github.com/akveo/ng2-admin/commit/2b25258))
* **dependencies:** fix versions ([62810ae](https://github.com/akveo/ng2-admin/commit/62810ae))
* **dependencies:** fix versions ([5523029](https://github.com/akveo/ng2-admin/commit/5523029))
* **dependencies:** update dependencies, fix [#694](https://github.com/akveo/ng2-admin/issues/694), fix [#695](https://github.com/akveo/ng2-admin/issues/695) ([08de1f6](https://github.com/akveo/ng2-admin/commit/08de1f6))
* **docker:** fix [#712](https://github.com/akveo/ng2-admin/issues/712) ([cd7520e](https://github.com/akveo/ng2-admin/commit/cd7520e))
* **documents:** fix [#771](https://github.com/akveo/ng2-admin/issues/771) ([78c70ae](https://github.com/akveo/ng2-admin/commit/78c70ae))
* **editors:** fix an issue where ckeditor doesn't load on the demo ([0a4f40c](https://github.com/akveo/ng2-admin/commit/0a4f40c))
* **forms:** bad linking correction between label and input ([#886](https://github.com/akveo/ng2-admin/issues/886)) ([bb6ecd3](https://github.com/akveo/ng2-admin/commit/bb6ecd3))
* **maps:** fix [#813](https://github.com/akveo/ng2-admin/issues/813) ([b414eaf](https://github.com/akveo/ng2-admin/commit/b414eaf))
* **menu:** fix inactive menu item ([2468710](https://github.com/akveo/ng2-admin/commit/2468710))
* **package:** add the description fields ([0d77147](https://github.com/akveo/ng2-admin/commit/0d77147))
* **sidebar:** fix [#997](https://github.com/akveo/ng2-admin/issues/997) ([0a3c98e](https://github.com/akveo/ng2-admin/commit/0a3c98e))
* **style:** fix [#958](https://github.com/akveo/ng2-admin/issues/958) ([#963](https://github.com/akveo/ng2-admin/issues/963)) ([06d1b98](https://github.com/akveo/ng2-admin/commit/06d1b98))
* **tables:** fix [#1000](https://github.com/akveo/ng2-admin/issues/1000) ([#1001](https://github.com/akveo/ng2-admin/issues/1001)) ([4b98e0e](https://github.com/akveo/ng2-admin/commit/4b98e0e))
* **tables:** fix [#700](https://github.com/akveo/ng2-admin/issues/700) ([c85eae7](https://github.com/akveo/ng2-admin/commit/c85eae7))
* **tables:** fix [#704](https://github.com/akveo/ng2-admin/issues/704) ([0fbad7e](https://github.com/akveo/ng2-admin/commit/0fbad7e))
* **theme:** fix cards background of blur theme ([0528e2e](https://github.com/akveo/ng2-admin/commit/0528e2e))
* **ui:** fix NgbModal declaration ([49ff10e](https://github.com/akveo/ng2-admin/commit/49ff10e))
* **validator:** change email regex to: http://emailregex.com/ ([4470607](https://github.com/akveo/ng2-admin/commit/4470607))


### Features

* **angular-cli:** integrate angular-cli :tada: ([#989](https://github.com/akveo/ng2-admin/issues/989)) ([7932421](https://github.com/akveo/ng2-admin/commit/7932421))
* **dependencies:** update yarn.lock file, fix debugging ([cc97bf1](https://github.com/akveo/ng2-admin/commit/cc97bf1))
* **layouts:** add file uploader ([#842](https://github.com/akveo/ng2-admin/issues/842)) ([7d03461](https://github.com/akveo/ng2-admin/commit/7d03461))
* **localization:** implement the localization using ngx-translate ([#830](https://github.com/akveo/ng2-admin/issues/830)) ([627d62d](https://github.com/akveo/ng2-admin/commit/627d62d))
* **login:** add the locale changes ([#993](https://github.com/akveo/ng2-admin/issues/993)) ([e033edf](https://github.com/akveo/ng2-admin/commit/e033edf))
* **test:** add testing infrastructure: karma, jasmine and chrome ([#982](https://github.com/akveo/ng2-admin/issues/982)) ([4420048](https://github.com/akveo/ng2-admin/commit/4420048))


### BREAKING CHANGES

* The webpack version of ng2-admin was moved to webpack [branch](https://github.com/akveo/ng2-admin/tree/webpack).
* We dropped the electron support as, unfortunately, angular-cli doesn't have the official support of electron. Please consider using preview 'webpack' (link to webpack branch) version manually.


### How to migrate

1. Pull the latest ng2-admin version from **https://github.com/akveo/ng2-admin.git**;
2. Copy your **src/app, src/assets, src/custom-typings.d.ts** from webpack version into **src** folder in cli version;
3. Add each of your required modules and styles into **.angular-cli.json**;
5. Change paths to file(images, fonts, ets);
6. Add **import * as $ from 'jquery';** into **app.component.ts** in cli version;

src/app/pages/pages.menu.ts
export const PAGES_MENU = [
  {
    path: 'pages',
    children: [
      {
        path: 'dashboard',
        data: {
          menu: {
            title: '仪表板',
            icon: 'ion-android-home',
            selected: false,
            expanded: false,
            order: 0
          }
        }
      },
      {
        path: 'organization',
        data: {
          menu: {
            title: '常用菜单',
            icon: 'ion-edit',
            selected: false,
            expanded: false,
            order: 100,
          }
        },
        children: [
          {
            path: 'ckeditor',
            data: {
              menu: {
                title: '常用菜单1',
                selected: false,
                expanded: false,
                order: 100,
              }
            }
          }
        ]
      },

      {
        path: '',
        data: {
          menu: {
            title: '组织管理',
            icon: 'ion-edit',
            selected: false,
            expanded: false,
            order: 100,
          }
        },
        children: [
          {
            path: 'department',
            data: {
              menu: {
                title: '部门管理',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'CreateDepartment',
                data: {
                  menu: {
                    title: '新建部门',
                  }
                }
              },
              {
                path: 'DepartmentList',
                data: {
                  menu: {
                    title: '部门列表',
                  }
                }
              },
              {
                path: 'test',
                data: {
                  menu: {
                    title: '测试',
                  }
                }
              },
            ]
          },
          {
            path: 'user',
            data: {
              menu: {
                title: '用户管理',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'usercreate',
                data: {
                  menu: {
                    title: '新增用户',
                  }
                }
              },
              {
                path: 'usermgmt',
                data: {
                  menu: {
                    title: '用户信息',
                  }
                }       
              },
            ]
          },
          {
            path: 'role',
            data: {
              menu: {
                title: '角色管理',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'rolecreate',
                data: {
                  menu: {
                    title: '创建角色',
                  }
                }
              },
              {
                path: 'roleprivilege',
                data: {
                  menu: {
                    title: '角色权限管理',
                  }
                }
              },
              {
                path: 'rolepeople',
                data: {
                  menu: {
                    title: '角色人员管理',
                  }
                }
              }
            ]
          },
          {
            path: 'dataprivilege',
            data: {
              menu: {
                title: '数据权限管理',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'DataOperatePrivilegeToRole',
                data: {
                  menu: {
                    title: '根据角色',
                  }
                }
              },
              {
                path: 'DataOperatePrivilegeToUser',
                data: {
                  menu: {
                    title: '根据用户',
                  }
                }
              }
            ]
          },
          {
            path: 'privilegereport',
            data: {
              menu: {
                title: '权限报表',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'UserPrivilegeReport',
                data: {
                  menu: {
                    title: '用户权限报表',
                  }
                }
              }
            ]
          },
          {
            path: 'log',
            data: {
              menu: {
                title: '日志',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'LogForUserLogin',
                data: {
                  menu: {
                    title: '用户登录日志',
                  }
                }
              },
              {
                path: 'LogForUserOperate',
                data: {
                  menu: {
                    title: '用户操作日志',
                  }
                }
              },
              {
                path: 'LogForUserLoginException',
                data: {
                  menu: {
                    title: '用户异常登录日志',
                  }
                }
              },
              {
                path: 'LogForUserOperateException',
                data: {
                  menu: {
                    title: '用户异常操作日志',
                  }
                }
              }
            ]
          },
          {
            path: 'chart',
            data: {
              menu: {
                title: '图表展示',
                selected: false,
                expanded: false,
                order: 100,
              }
            },
            children: [
              {
                path: 'basic',
                data: {
                  menu: {
                    title: '多种图表',
                  }
                }
              }
            ]
          },
        ]
      },

      {
        path: 'dashboard',
        data: {
          menu: {
            title: 'general.menu.dashboard',
            icon: 'ion-android-home',
            selected: false,
            expanded: false,
            order: 0
          }
        }
      },
      {
        path: 'editors',
        data: {
          menu: {
            title: 'general.menu.editors',
            icon: 'ion-edit',
            selected: false,
            expanded: false,
            order: 100,
          }
        },
        children: [
          {
            path: 'ckeditor',
            data: {
              menu: {
                title: 'general.menu.ck_editor',
              }
            }
          }
        ]
      },

      {
        path: 'components',
        data: {
          menu: {
            title: 'general.menu.components',
            icon: 'ion-gear-a',
            selected: false,
            expanded: false,
            order: 250,
          }
        },
        children: [
          {
            path: 'treeview',
            data: {
              menu: {
                title: 'general.menu.tree_view',
              }
            }
          }
        ]
      },
      {
        path: 'charts',
        data: {
          menu: {
            title: 'general.menu.charts',
            icon: 'ion-stats-bars',
            selected: false,
            expanded: false,
            order: 200,
          }
        },
        children: [
          {
            path: 'chartist-js',
            data: {
              menu: {
                title: 'general.menu.chartist_js',
              }
            }
          }
        ]
      },
      {
        path: 'ui',
        data: {
          menu: {
            title: 'general.menu.ui_features',
            icon: 'ion-android-laptop',
            selected: false,
            expanded: false,
            order: 300,
          }
        },
        children: [
          {
            path: 'typography',
            data: {
              menu: {
                title: 'general.menu.typography',
              }
            }
          },
          {
            path: 'buttons',
            data: {
              menu: {
                title: 'general.menu.buttons',
              }
            }
          },
          {
            path: 'icons',
            data: {
              menu: {
                title: 'general.menu.icons',
              }
            }
          },
          {
            path: 'modals',
            data: {
              menu: {
                title: 'general.menu.modals',
              }
            }
          },
          {
            path: 'slim',
            data: {
              menu: {
                title: 'Slim loading bar',
              }
            }
          },
          {
            path: 'grid',
            data: {
              menu: {
                title: 'general.menu.grid',
              }
            }
          },
        ]
      },
      {
        path: 'forms',
        data: {
          menu: {
            title: 'general.menu.form_elements',
            icon: 'ion-compose',
            selected: false,
            expanded: false,
            order: 400,
          }
        },
        children: [
          {
            path: 'inputs',
            data: {
              menu: {
                title: 'general.menu.form_inputs',
              }
            }
          },
          {
            path: 'layouts',
            data: {
              menu: {
                title: 'general.menu.form_layouts',
              }
            }
          }
        ]
      },
      {
        path: 'tables',
        data: {
          menu: {
            title: 'general.menu.tables',
            icon: 'ion-grid',
            selected: false,
            expanded: false,
            order: 500,
          }
        },
        children: [
          {
            path: 'basictables',
            data: {
              menu: {
                title: 'general.menu.basic_tables',
              }
            }
          },
          {
            path: 'smarttables',
            data: {
              menu: {
                title: 'general.menu.smart_tables',
              }
            }
          },
          {
            path: 'datatables',
            data: {
              menu: {
                title: 'Data Tables',
              }
            }
          },
          {
            path: 'hottables',
            data: {
              menu: {
                title: 'Hot Tables',
              }
            }
          }
        ]
      },
      {
        path: 'tabletest',
        data: {
          menu: {
            title: 'tabletest',
            icon: 'ion-ios-location-outline',
            selected: false,
            expanded: false,
            order: 600,
          }
        },
        children: [
          {
            path: 'table1',
            data: {
              menu: {
                title: 'table1',
              }
            }
          }
        ]
      },
      {
        path: 'maps',
        data: {
          menu: {
            title: 'general.menu.maps',
            icon: 'ion-ios-location-outline',
            selected: false,
            expanded: false,
            order: 600,
          }
        },
        children: [
          {
            path: 'googlemaps',
            data: {
              menu: {
                title: 'general.menu.google_maps',
              }
            }
          },
          {
            path: 'leafletmaps',
            data: {
              menu: {
                title: 'general.menu.leaflet_maps',
              }
            }
          },
          {
            path: 'bubblemaps',
            data: {
              menu: {
                title: 'general.menu.bubble_maps',
              }
            }
          },
          {
            path: 'linemaps',
            data: {
              menu: {
                title: 'general.menu.line_maps',
              }
            }
          }
        ]
      },
      {
        path: '',
        data: {
          menu: {
            title: 'general.menu.pages',
            icon: 'ion-document',
            selected: false,
            expanded: false,
            order: 650,
          }
        },
        children: [
          {
            path: ['/login'],
            data: {
              menu: {
                title: 'general.menu.login'
              }
            }
          },
          {
            path: ['/register'],
            data: {
              menu: {
                title: 'general.menu.register'
              }
            }
          }
        ]
      },
      {
        path: '',
        data: {
          menu: {
            title: 'general.menu.menu_level_1',
            icon: 'ion-ios-more',
            selected: false,
            expanded: false,
            order: 700,
          }
        },
        children: [
          {
            path: '',
            data: {
              menu: {
                title: 'general.menu.menu_level_1_1',
                url: '#'
              }
            }
          },
          {
            path: '',
            data: {
              menu: {
                title: 'general.menu.menu_level_1_2',
                url: '#'
              }
            },
            children: [
              {
                path: '',
                data: {
                  menu: {
                    title: 'general.menu.menu_level_1_2_1',
                    url: '#'
                  }
                }
              }
            ]
          }
        ]
      },
      {
        path: '',
        data: {
          menu: {
            title: 'general.menu.external_link',
            url: 'http://akveo.com',
            icon: 'ion-android-exit',
            order: 800,
            target: '_blank'
          }
        }
      }
    ]
  }
];
